radicle-node/src/tests.rs

```diff
@@ -600,6 +600,61 @@ fn test_refs_announcement_relay() {
     );
 }
 
+#[test]
+fn test_refs_announcement_fetch() {
+    logger::init(log::Level::Debug);
+
+    let tmp = tempfile::tempdir().unwrap();
+    let mut alice = Peer::config(
+        "alice",
+        [7, 7, 7, 7],
+        Storage::open(tmp.path().join("alice")).unwrap(),
+        peer::Config::default(),
+    );
+    let eve = Peer::config(
+        "eve",
+        [8, 8, 8, 8],
+        Storage::open(tmp.path().join("eve")).unwrap(),
+        peer::Config::default(),
+    );
+
+    let bob = {
+        let mut rng = fastrand::Rng::new();
+        let signer = MockSigner::new(&mut rng);
+        let storage = fixtures::storage(tmp.path().join("bob"), &signer).unwrap();
+
+        Peer::config(
+            "bob",
+            [9, 9, 9, 9],
+            storage,
+            peer::Config {
+                signer,
+                rng,
+                ..peer::Config::default()
+            },
+        )
+    };
+    let bob_inv = bob.inventory();
+
+    alice
+        .track_repo(&bob_inv[0], tracking::Scope::Trusted)
+        .unwrap();
+
+    alice.connect_to(&bob);
+    alice.connect_to(&eve);
+
+    alice.receive(eve.id(), Message::Subscribe(Subscribe::all()));
+
+    // Alice receives refs_announcement from Bob.
+    alice.receive(bob.id(), bob.refs_announcement(bob_inv[0]));
+
+    assert_matches!(
+        alice.messages(bob.id()).next(),
+        Some(Message::Fetch { rid }) if rid == bob_inv[0],
+        "A fetch is sent to Bob as this is a new repo for Alice"
+    );
```

Comment:
I'm not sure what we're testing here.. Would this test fail without the changes you made? Afaict it would pass because we were always fetching previously. If so, we should additionally check that Alice doesn't fetch based on Eve's announcement no?

Reply:
This test only covers the basic positive case, which indeed was covered by existing other tests as well. I will remove this test.